### PR TITLE
Add functionality for guest user accounts

### LINKED ISSUE #7: Create guest user system
Users should be able to use the app without logging in (at the cost of not having their progress saved). Users should be able to create an account in order to save their current progress.

### PR DESCRIPTION
**Pull request by BusinessJoe**
This adds the ability to create a guest user account via the `login_guest` endpoint.

A guest account has all the functionality of a regular account. If someone creates an account while signed in as a guest, the guest account will be converted into a regular account so that their progress isn't lost. Guest accounts older than 1 week will be automatically deleted. 

Closes #7.

### CODE REVIEW

File: backend/intervals/settings.py [hunk 1]
```diff
@@ -124,7 +127,7 @@
 
 USE_L10N = True
 
-USE_TZ = True
+# USE_TZ = True
```
Comment (JJamali) [inline backend/intervals/settings.py]: Does this need to be commented?

File: backend/intervals/settings.py [hunk 2]
```diff
@@ -142,3 +145,12 @@
 }
 
 
+CELERY_BROKER_URL = "redis://redis:6379"
+CELERY_RESULT_BACKEND = "redis://redis:6379"
+
+CELERY_BEAT_SCHEDULE = {
+    "clear_old_guests": {
+        "task": "intervals.tasks.clear_old_guests",
+        "schedule": crontab(minute="*/1"),
```
Comment (JJamali) [inline backend/intervals/settings.py]: Maybe we should include a comment here explaining the schedule.

File: backend/intervalsApp/apps.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 from django.apps import AppConfig
+from datetime import timedelta
 
 
 class IntervalsAppConfig(AppConfig):
     name = 'intervalsApp'
     SCORE_RANGE = 20  # The number of most recent results factor into calculation of score
     TOTAL_LEVELS = 5  # Maximum level. The first level is level 0.
+    GUEST_INACTIVE_LIMIT = timedelta(weeks=1)  # Minimum time a guest can be inactive for before they are deleted
```
Comment (JJamali) [inline backend/intervalsApp/apps.py]: Add newline at end of file.

File: backend/intervalsApp/views.py [hunk 4]
```diff
@@ -33,11 +35,39 @@ class UserList(generics.ListCreateAPIView):
...
+                login(request, user)
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class CreateGuest(generics.CreateAPIView):
+    serializer_class = BriefUserSerializer
+
+    def create(self, request, *args, **kwargs):
+        random_id = str(uuid.uuid4())
```
Comment (JJamali) [inline backend/intervalsApp/views.py:60]: Might be a good idea to comment on how the uuid functions / why it is used.

File: backend/intervalsApp/views.py [hunk 5]
```diff
@@ -174,14 +204,19 @@ def login_view(request):
...
         user = authenticate(request, username=username, password=password)
-        print(user)
         if user is not None:
             login(request, user)
             return Response(status=status.HTTP_200_OK)
         else:
             return Response(status=status.HTTP_401_UNAUTHORIZED)
 
 
+@api_view(['POST'])
```
Comment (JJamali) [inline backend/intervalsApp/views.py]: Should leave a comment explaining why this passes if POST

File: docker-compose.yaml [hunk 6]
```diff
@@ -41,4 +44,38 @@ services:
...
+      - redis
+
+  db:
+    image: postgres
+    volumes:
+      - ./data/db:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_DB=postgres
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
```
Comment (JJamali) [inline docker-compose.yaml]: Add newline

**Review by JJamali — CHANGES_REQUESTED** [review]

File: backend/intervals/settings.py (hunk 2 above)
Comment (BusinessJoe) [reply backend/intervals/settings.py]: Yeah definitely. It also looks like I set the task to run every minute instead of each day. (Whoops)

**Pull request merged**